### PR TITLE
Migrate to KCL V2

### PR DESCRIPTION
**Pull request by gheine**

### CODE REVIEW

File: app/io/flow/event/v2/KinesisConsumer.scala [hunk 1]
```diff
@@ -41,9 +51,10 @@ case class KinesisConsumer (
     .withKeyValue("worker_id", config.workerId)
     .info("Started")
 
-  exec.execute(worker)
+  exec.execute(scheduler)
 
   def shutdown(): Unit = {
+    scheduler.shutdown()
```
Comment (nataliab) [inline app/io/flow/event/v2/KinesisConsumer.scala]: Should we implement `graceful shutdown`? `scheduler.startGracefulShutdown() ...`

File: app/io/flow/event/v2/KinesisConsumer.scala [hunk 2]
```diff
@@ -141,14 +152,20 @@ case class KinesisRecordProcessor[T](
...
+    handleCheckpoint(input.checkpointer)
   }
 
-  private def handleCheckpoint(checkpointer: IRecordProcessorCheckpointer, retries: Int = 0): Unit = {
+  override def leaseLost(input: LeaseLostInput): Unit = {
+    logger_.withKeyValue("reason", "leaseLost").info("Shutting down")
+  }
+
+  override def shardEnded(input: ShardEndedInput): Unit = {
+    logger_.withKeyValue("reason", "shardEnded").info("Shutting down")
```
[the remark this answers is not in the document]
Comment (gheine) [reply app/io/flow/event/v2/KinesisConsumer.scala]: 👍

File: app/io/flow/event/v2/KinesisConsumer.scala [hunk 3]
```diff
@@ -141,14 +152,20 @@ case class KinesisRecordProcessor[T](
     }
   }
 
-  override def shutdown(input: ShutdownInput): Unit = {
-    logger_.withKeyValue("reason", input.getShutdownReason.toString).info("Shutting down")
-    if (input.getShutdownReason == ShutdownReason.TERMINATE) {
-      handleCheckpoint(input.getCheckpointer)
-    }
+  override def shutdownRequested(input: ShutdownRequestedInput): Unit = {
```
[the remark this answers is not in the document]
Comment (gheine) [reply app/io/flow/event/v2/KinesisConsumer.scala]: 👍

File: app/io/flow/event/v2/KinesisConsumer.scala (hunk 1 above)
Comment (gheine) [reply app/io/flow/event/v2/KinesisConsumer.scala]: 👍

File: app/io/flow/event/v2/KinesisConsumer.scala [hunk 4]
```diff
@@ -87,35 +103,35 @@ case class KinesisRecordProcessor[T](
...
     if (kinesisRecords.nonEmpty) {
       val flowRecords = kinesisRecords.map { record =>
-        val buffer = record.getData
+        val buffer = record.data
         val bytes = Array.fill[Byte](buffer.remaining)(0)
         buffer.get(bytes)
 
         Record.fromByteArray(
-          arrivalTimestamp = new DateTime(record.getApproximateArrivalTimestamp),
+          arrivalTimestamp = new DateTime(record.approximateArrivalTimestamp),
```
Comment (mikeclimbrock) [inline app/io/flow/event/v2/KinesisConsumer.scala]: this caused a production service-wide issue in consuming events:
```
"java.lang.IllegalArgumentException: No instant converter found for type: java.time.Instant
\tat org.joda.time.convert.ConverterManager.getInstantConverter(ConverterManager.java:166)
\tat org.joda.time.base.BaseDateTime.<init>(BaseDateTime.java:171)
\tat org.joda.time.DateTime.<init>(DateTime.java:257)
\tat io.flow.event.v2.KinesisRecordProcessor.$anonfun$processRecords$2(KinesisConsumer.scala:124)
```

full stack trace can be seen in Sumo.

**Comment by jlogeart** [comment]
Note (one of?) the culprit for this change not to work is the removal of the dynamodb table name keeping track of the checkpoints: https://github.com/flowcommerce/lib-event/pull/476/files#diff-4307996d179b172eb3a13d03350fe238L82

All consumers from one app are now checkpointing in the same `appName` table, even if they consume multiple streams, leading to inconsistent behavior: https://console.aws.amazon.com/dynamodb/home?region=us-east-1#tables:

Screen Shot 2019-08-07 at 10 00 06 AM


Integration tests have been added and this PR reintroduces the right table: https://github.com/flowcommerce/lib-event/pull/488

**Pull request merged**